cirq-core/cirq/ops/gate_operation.py

```diff
@@ -201,6 +201,18 @@ def _mixture_(self) -> Sequence[Tuple[float, Any]]:
             return getter()
         return NotImplemented
 
+    def _has_channel_(self) -> bool:
+        getter = getattr(self.gate, '_has_channel_', None)
```

Comment:
should we throw a deprecation warning in these methods as well?

Reply:
The deprecation behavior gets inherited because this doesn't forward to `_kraus_`. There's a test checking it.

Reply:
I did consider and try this, but the `cirq.kraus` and `cirq.has_kraus` protocols fall back to these methods in expressions such as `cirq.kraus(cirq.SingleQubitGate())`. This causes test failures since we do not allow cirq's internal codepaths to raise deprecation warnings.

Reply:
Also, what Craig said. More precisely, we don't need it because of what Craig said and we don't want it because of what I said.

BTW: the reason the inheriting behavior Craig alluded to does not break tests is that in tests `getattr` returns `None`, so the propagating `_channel_` does not end up calling any marked-as-deprecated `_channel_`.